### DISCUSSION TITLE
PHP errors

### DISCUSSION THREAD

**Discussion by RosieCallan91** [bootscore/bootscore#501] [Q&A]
Good morning everyone,

I have recently taken on the task of making a wordpress site responsive for a college. I am using the Bootscore theme and have done all the work I want to do on the site. I have no prior experience of Wordpress and now unfortunately I am getting PHP errors 
- Warning: file_put_contents(/code/wp-content/themes/bootscore-main/css/main.css): failed to open stream: Permission denied in /code/wp-content/themes/bootscore-main/inc/scss-compiler.php on line 58 Warning: file_put_contents(/code/wp-content/themes/bootscore-main/css/main.map): failed to open stream: Permission denied in /code/wp-content/themes/bootscore-main/inc/scss-compiler.php on line 60
The site url is - https://dev-institute-of-technology-north-west.pantheonsite.io/

Any advice would be really greatly appreciated.

Many thanks

Rosie

**Comment by crftwrk** [comment]
Hi Rosie,

I see actually no PHP error on your site?

**Reply by RosieCallan91** [reply]
Thanks for getting back to me. My bad I posted my dev url but it's only showing on test - https://test-institute-of-technology-north-west.pantheonsite.io/

**Reply by crftwrk** [reply]
Is this the same server? If not, it seems to be a server setting `Permission denied in...` and the theme's `main.css` isn't created. Transfer your site to a different server or check the writing access on actual server should solve this. You may ask your hoster to change writing settings.

As you see, on dev-site it's working fine. So, that isn't something with the theme.

**Comment by RosieCallan91** [comment]
The hoster seemed to think it was something to do with how the themes writing in test and live as its read only
Thank you

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by RosieCallan91** [comment]
I have been in a lengthy chat with Pantheon today and have been trying to create a symlink but it doesn't allow me to. They believe it's something to do with the theme. Can anyone possibly help with this. Many thanks

**Reply by crftwrk** [reply]
Hi Rosie,

you have two installations of your site:

| Fine  | Error |
| ------------- | ------------- |
| https://dev-institute-of-technology-north-west.pantheonsite.io  | https://test-institute-of-technology-north-west.pantheonsite.io  |
| <img width="361" alt="Bildschirmfoto 2023-06-07 um 19 21 43" src="https://github.com/bootscore/bootscore/assets/51531217/18b568ce-a9a6-4991-9f17-90c2c692b3c8">  | <img width="399" alt="Bildschirmfoto 2023-06-07 um 19 21 28" src="https://github.com/bootscore/bootscore/assets/51531217/0b7bbb35-74b4-4d35-9a26-cc24e15d16e8">  |

As you can see on the left screenshot, there is folder `css` and file `main.css`. Both does not exists on the error screenshot on the right. 

If you have a quick look to the [themes folder](https://github.com/bootscore/bootscore), you will see that `css` folder and `main.css` file does not exists in bootScore as well. Both are created by scssphp compiler on first installation or if one of the SCSS file changed and `main.css` is the file where all the styles stored.

The PHP error says that it's not possible to create this file, I guess this is because scssphp has no writing access on this server installation.

What you can do:

Try to do keep things as simple as possible.

- Log in via ftp to the fine site, go to theme folder and download folder `css`. Upload this folder to the error page theme root folder.
- Download the entire `bootscore-main` theme folder from fine site, zip and upload it in the backend to error page.
- Install a copy of the fine site on a different server. 

If both sides are identical, it can only be the server.

Hope that helps

**Comment by RosieCallan91** [comment]
Hi there,

Thanks for looking at this but I can download any files to my test site due to my server not allowing this for test and live. I've tried creating Symlinks and they say that is the only way forward with it. To no avail. When you connect with Filezilla to dev the css file is there with the main.css and man.map. When you git clone the code them 2 files are there but not within a folder of css, CSS doesn't exist. If I mkdir css and mv the 2 files into it. I git status and it says I have deleted the 2 files. I'm thinking my only resolution for this is to change my theme? I've literally spent days trying to resolve this now. Thank you Rosie

**Reply by crftwrk** [reply]
Hi Rosie,

> I've literally spent days trying to resolve this now.

as I mentioned in my previous comment "Try to do keep things as simple as possible" means that you should stop wasting your time figuring this out on the broken page and use your theme from the working site instead. 

- Simply connect the [working page](https://dev-institute-of-technology-north-west.pantheonsite.io/) with Filezilla
- Download the entire `bootscore-main` theme folder  
- zip folder
- Go to broken page's backend to Appearance > Themes > Add new, select zip file and upload it
- Done

> I'm thinking my only resolution for this is to change my theme?

Well, bootScore is a coding theme. Means there are no settings in the backend and all changes have to be done by SCSS, HTML, PHP and JS. Basic knowledge in this languages and basic knowledge how WordPress works is required. 

If you have this skills, then bootScore is a time-saving and flexible theme which allows you to create whatever you want. Everything is written down in the [docs](https://bootscore.me/documentation/), it's simple to learn.

So, maybe a theme with simple CSS instead the SCSS feature should be easier. But as I said in all my previous comments, this is a server thing because compiler has no writing access to create this folder.
